.travis.yml

```diff
@@ -12,6 +12,16 @@ matrix:
         - BUILD_APIDOC=true
         # Auth token to push API documentation
         - secure: "JKeXk8p65hodb12PVRST6A90swsNubc+46EbSJGSghldIxbFWLBAlwU+KLeOMO4V0veu6k4lnMa50V0UYFZmoUsS6W0aL5Ybo98SpzXHiNLOmOluoqJoF9TBsOTCCRFbWbccgJyVEtulgRcdml96naS51lq9Sw/VO/N3Z472304="
+    - os: linix
+      language: python
+      python: 2.7.14
+      install:
```

Comment:
I would rather prefer to move the `install`, `script` and `before_*` sections out of the build matrix and add them rather as conditional steps below.

But of now it can stay as it is.

Reply:
My IMHO that `install`, `scripts` and `before_*` should not be global and present in matrix for each entry, but I did not know how to reuse code properly, may be put `*.sh` script to repository.